### PR TITLE
Add context menu

### PR DESCRIPTION
**Pull request by henriettemoe**
Elements in grid will now have their own context menu where tools like edit and delete will be placed. The delete tool is added for boxes, so they can now be deleted.

### CODE REVIEW

File: src/components/ContextMenu/ContextMenu.module.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  top: -0.45em;
+  transform: translateY(-100%);
+  white-space: nowrap;
+  user-select: none;
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  background-color: #c4e3f7;
+  border: 1px solid #09f;
+  border-radius: 2px;
+  -webkit-filter: drop-shadow(0px 0px 0.25em rgba(44, 44, 44, 0.75));
```
Comment (boyum) [inline src/components/ContextMenu/ContextMenu.module.scss:13]: `-webkit-filter` is not needed, as all browsers we support already support `filter`. https://caniuse.com/css-filters

File: src/components/ContextMenu/ContextMenu.module.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    position: absolute;
+    display: block;
+    content: "";
+    border-style: solid;
+    border-width: 0.6em 0.6em 0 0;
+    border-color: #c4e3f7 transparent transparent transparent;
+    bottom: calc(-0.6em + 2px);
+    left: 0;
+    z-index: 1;
+  }
```
Comment (boyum) [inline src/components/ContextMenu/ContextMenu.module.scss:38]: We should perhaps explain why `bottom` is the way it is. Maybe a `$border-width` variable might work.

```suggestion
  &::after {
    $border-width: 0.6em;
  
    position: absolute;
    display: block;
    content: "";
    border-style: solid;
    border-width: $border-width $border-width 0 0;
    border-color: #c4e3f7 transparent transparent transparent;
    bottom: calc(-$border-width + 2px);
    left: 0;
    z-index: 1;
  }
```

File: src/icons.tsx [hunk 3]
```diff
@@ -19,6 +22,12 @@ export const Icons: React.FC<IconsProps> = ({ icon, className }) => {
   if (icon === ToolbarButtonType.CreateArrow) {
     return <CreateArrow className={className} />;
   }
+  if (icon === ContextMenuButtonType.Edit) {
```
Comment (boyum) [inline src/icons.tsx:25]: `Icons` could PERHAPS be re-written as 

```tsx
const allIcons = {
  [ToolbarButtonType.MapColor]: <MapColor className={className} />,
  [ToolbarButtonType.CreateBox]: <CreateBox className={className} />,
  [ToolbarButtonType.CreateArrow]: <CreateArrow className={className} />,
  [ToolbarButtonType.Edit]: <Edit className={className} />,
  [ToolbarButtonType.Delete]: <Delete className={className} />,
};


export const Icons: React.FC<IconsProps> = ({ icon, className }) => {
  const defaultIcon = <MapColor className={className} />;
  
  return allIcons[icon] ?? defaultIcon;
};
```

**Review by boyum — APPROVED** [review]
Works great ✨  We need to find out how to open the edit view.

File: src/icons.tsx (hunk 3 above)
Comment (boyum) [reply src/icons.tsx:25]: (and be called "Icon"?)

Comment (henriettemoe) [reply src/icons.tsx:25]: How will className be included? Maybe have the change with icons in a new task?

**Pull request merged**